addons/maptools/CfgVehicles.hpp

```diff
@@ -43,17 +43,150 @@ class CfgVehicles {
                     exceptions[] = {"isNotDragging", "notOnMap", "isNotInside", "isNotSitting"};
                     showDisabled = 1;
                 };
-                class ACE_MapToolsAlignNorth {
-                    displayName = CSTRING(MapToolsAlignNorth);
+                class ACE_MapToolsAlign {
+                    displayName = CSTRING(AlignTo);
                     condition = QUOTE(GVAR(mapTool_Shown) != 0);
-                    statement = QUOTE(GVAR(mapTool_angle) = 0;);
+                    statement = "";
+                    exceptions[] = {"isNotDragging", "notOnMap", "isNotInside", "isNotSitting"};
+                    showDisabled = 0;
+                    class ACE_MapToolsAlignToPlottingBoardRuler {
+                        displayName = CSTRING(ToPlottingBoardRulerLabel);
+                        condition = QUOTE(GVAR(mapTool_Shown != 0) && GVAR(plottingBoard_Shown == 2));
+                        statement = QUOTE(GVAR(mapTool_angle) = GVAR(plottingBoard_rulerAngle));
+                        exceptions[] = {"isNotDragging", "notOnMap", "isNotInside", "isNotSitting"};
+                        showDisabled = 1;
+                    };
+                    class ACE_MapToolsAlignToPlottingBoardAcrylic {
+                        displayName = CSTRING(ToPlottingBoardAcrylicLabel);
+                        condition = QUOTE(GVAR(mapTool_Shown != 0) && GVAR(plottingBoard_Shown != 0));
+                        statement = QUOTE(GVAR(mapTool_angle) = GVAR(plottingBoard_acrylicAngle));
+                        exceptions[] = {"isNotDragging", "notOnMap", "isNotInside", "isNotSitting"};
+                        showDisabled = 1;
+                    };
+                    class ACE_MapToolsAlignToPlottingBoard {
+                        displayName = CSTRING(ToPlottingBoardLabel);
+                        condition = QUOTE(GVAR(mapTool_Shown != 0) && GVAR(plottingBoard_Shown != 0));
+                        statement = QUOTE(GVAR(mapTool_angle) = GVAR(plottingBoard_angle));
+                        exceptions[] = {"isNotDragging", "notOnMap", "isNotInside", "isNotSitting"};
+                        showDisabled = 1;
+                    };
+                    class ACE_MapToolsAlignCompass {
+                        displayName = CSTRING(ToCompassLabel);
+                        condition = QUOTE(GVAR(mapTool_Shown) != 0 && {getUnitLoadout ACE_player param [ARR_2(9, [])] param [ARR_2(3, '')] != ''});
```

Comment:
1) 609 seems like a weird magic number.

2) Original Code that I never touched: https://github.com/acemod/ACE3/blob/f774eaf787f32670926259f470abd5b0303d0357/addons/maptools/CfgVehicles.hpp#L55

Reply:
609 is like a weird magic number, however it's defined on the [wiki page](https://community.bistudio.com/wiki/getSlotItemName). All of these numbers could be added to e.g. `.hpp` in `common` so that all other components can easily use them.
Imo might as well update the code to something that is surely faster.

Reply:
It's a drive-by change, `getSlotItemName` was introduced in 2.14

Reply:
> It's a drive-by change, `getSlotItemName` was introduced in 2.14

So should I make another PR to change it?

Reply:
If you'd like to. I was talking to LorenLuke via DMs as to reasoning behind the change and left a comment to document it. 

Handling it here or elsewhere won't change anything.

Reply:
> If you'd like to. I was talking to LorenLuke via DMs as to reasoning behind the change and left a comment to document it.
> 
> Handling it here or elsewhere won't change anything.

Changing it in this PR would make the most sense imo.